collect_app/src/main/java/org/odk/collect/android/activities/GeoPointMapActivity.java

```diff
@@ -496,6 +504,14 @@ private void showGPSDisabledAlertToUser() {
         errorDialog.show();
     }
 
+    private void resetTrashButton() {
```

Comment:
If the trash button has to be modified whenever `marker` is updated, so then they should sit close to each other. That means, `marker` should be modified only from a single method which should also contain the logic to update the `trash` icon along with it every time. This will remove the overhead of updating the state of trash icon from all of the places. 
What are your thoughts?

Reply:
I understand what you mean, let me take look at it.